entities/container.py

```diff
@@ -12,6 +17,63 @@ def __init__(self, env, region):
 
         self.region = region
         self.location = region.draw_location()
+        self.id = next(self._ids)
+        self.state = ContainerState.EMPTY
+
+
+    def create_bids(self, best_shipments, registrationkey):
+        containerbid = namedtuple('containerbid', 'container_registration_key shipment_registration_key biddingvalue')
+        containerbids = []
+        for entry in best_shipments:
+            shipment_registration_key = entry[0]
+            shipment = entry[2]
+            transport_cost_empty = route_euclidean_distance(self.env, self.location, shipment.location)
+            transport_cost_shipment = route_euclidean_distance(self.env, shipment.location, shipment.destination)
+            biddingvalue = transport_cost_empty + transport_cost_shipment
+            #TODO add additional profit margin for container or decrease of biddingvalue based on urgency
+            container_bid = containerbid(container_registration_key = registrationkey,
+                                        shipment_registration_key = shipment_registration_key,
+                                        biddingvalue = biddingvalue)
+            containerbids.append(container_bid)
+        return containerbids
+
+    def request_shipments(self):
+        #TODO add functionality of selecting shipments from auctions outside current region
+        if EntityTypes.__members__['SHIPMENT'] not in \
```

Comment:
Don't use `__member__`. Just do `EntityTypes.SHIPMENT`

Reply:
when the key has not been added to the dictionary, using EntityTypes.SHIPMENT gives an error, that's why i used __member__

Reply:
That is simply not true. Doing `EntityTypes.SHIPMENT not in self.region.auctioneer.entities.keys()` is perfectly valid python, and should just work. Is there something else wrong?

In any case, do not use `__members__` for this: you don't need that field (ever).

Reply:
Seemed that previously something else was indeed wrong, I just changed it succesfully